sql/ddl_log.cc

```diff
@@ -3029,13 +3029,15 @@ static bool ddl_log_write(DDL_LOG_STATE *ddl_state,
   error= ((ddl_log_write_entry(ddl_log_entry, &log_entry)) ||
           ddl_log_write_execute_entry(log_entry->entry_pos,
                                       &ddl_state->execute_entry));
-  mysql_mutex_unlock(&LOCK_gdl);
+  DBUG_EXECUTE_IF("ddl_log_write_fail", error= true;);
   if (error)
   {
     if (log_entry)
       ddl_log_release_memory_entry(log_entry);
+    mysql_mutex_unlock(&LOCK_gdl);
     DBUG_RETURN(1);
   }
+  mysql_mutex_unlock(&LOCK_gdl);
```

Comment:
I was going to suggest to a comment that explains why `ddl_log_release_memory_entry()` needs to be protected by the mutex. Even better, add `mysql_mutex_assert_owner(&LOCK_gdl)` to the beginning of that function. But, I now see that the function already contains such an assertion. So, this is obviously correct and does not need a comment. This error handling code path was previously uncovered in any tests.